server.go

```diff
@@ -0,0 +1,19 @@
+package proxy
+
+import (
+	"context"
+	"net"
+
+	"github.com/getlantern/errors"
+)
+
+// Serve runs a proxy server using the given Listener
+func (proxy *proxy) Serve(l net.Listener) error {
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			return errors.New("Unable to accept: %v", err)
+		}
+		go proxy.Handle(context.Background(), conn)
```

Comment:
The Proxy interface is public and we want to allow other server implementations to pass their own context. This server is just a convenience, not the only entry point.